src/org/httpkit/sni_client.clj

```diff
@@ -5,41 +5,50 @@
   (:import [java.net URI]
            [javax.net.ssl SNIHostName SSLEngine SSLParameters]))
 
-(defn ssl-configurer
-  "SNI-capable SSL configurer.
-  May be used as an argument to `org.httpkit.client/make-client`:
-    (make-client :ssl-configurer (ssl-configurer))"
-  [^SSLEngine ssl-engine ^URI uri]
-  (let [host-name  (SNIHostName. (.getHost uri))
-        ssl-params (doto (.getSSLParameters ssl-engine)
-                     (.setServerNames [host-name]))]
-    (doto ssl-engine
-      (.setUseClientMode true) ; required for JVM 12/13 but not for JVM 8
-      (.setSSLParameters ssl-params))))
+(defn- parse-java-version
+  "Ref. https://stackoverflow.com/a/2591122"
+  [^String s]
+  (if (.startsWith s "1.") ; e.g. "1.6.0_23"
+    (Integer/parseInt (.substring s 2 3))
+    (let [dot-idx (.indexOf s ".")] ; e.g. "9.0.1"
```

Comment:
I guess this is fine, but you can also `(-> s (str/split #"/.") first Integer/parseInt)`.